src/deposit.cairo

```diff
@@ -177,15 +213,35 @@ mod Deposit {
                 );
         }
 
+        /// Closes position on ZKlend.
+        ///
+        /// # Panics
+        ///
+        /// `is_position_open` storage variable is set to false('Open position not exists')
+        /// `supply_price` or `debt_price` is `0`
+        /// address of the caller is not equal to `owner` storage address
+        ///
+        /// # Paraemters
+        /// * `supply_token`: ContractAddress - Address of the token used as collateral.
+        /// * `debt_token`: ContractAddress - Address of the token used as borrowing.
+        /// * `pool_key`: PoolKey - Ekubo type for obtaining info about the pool and swapping
+        /// tokens.
+        /// * `supply_price`: felt252 - Price of `supply` token in terms of `debt` token in Ekubo
```

Comment:
Is it exactly just price? How is that expressed? It must be a fraction or fixed point number right; what kind is it?

Reply:
And – can you give an example where supply_price is > 2**128? (I don't think it can reach that high?)

This format is used in Ekubo? (I don't think so)

I don't think it makes sense to have the price be in relation to one "full" (human-readable) unit; then you need to fiddle with decimals.

So 2 questions:

1. Why do you need the price? (Future slippage calculation?)
2. Don't you get the price from Ekubo after executing a swap?

Reply:
3. Why aren't you using the same format as Ekubo (since that's where you interact with the price) for storing the price? (You will then be able to truly get rid of decimals:) (I understand the Ekubo math is very hard and it might very well be a good reason)

Reply:
1. Calculation of how much can I borrow from zkLend, we will borrow 80%(I will try and test lower value) of allowed amount so precise median price from the oracles as used in zkLend is not required I suppose.
2. It's not given exactly on swap.  However I can get pool price from Ekubo as sqrt_ratio and then convert to the normal price. I am concerned about price of base token in terms of quote token(so ETH amount for USDC in ETH/USDC) and I wanted to calculate it on backend with normal decimals, but I should be able to handle it to depend on external params less.
3. Ekubo using prices as ticks or sqrt_ratio and zkLend as normal amounts, so there is a need for conversions unfortunately.